### ISSUE TITLE
Error in gen_fvect(): Cannot allocate memory, with low-frequency Nenufar data

### ISSUE THREAD

**Issue by alex88ridolfi** [scottransom/presto#135]
Hi Scott,

We are trying to use PRESTO to create dedispersed time series of some GC observations taken with Nenufar.

However, we are likely having problems with the memory allocation.
We suspect that this is due to the extremely low frequencies (45-82 MHz), which make the dedispersion challenging.

Below I am pasting the error I get with prepdata, when trying to create dedispersed time series from a filterbank file of a test observation, with the following characteristics:

```
aridolfi@nancep5:/data/aridolfi/B0950_test$ readfile B0950+08_digifilled.fil
Assuming the data is a SIGPROC filterbank file.

1: From the SIGPROC filterbank file 'B0950+08_digifilled.fil':
                  Telescope = Nancay
                Source Name = B0950+08
            Obs Date String = 2020-05-10T18:01:00.0282929
             MJD start time = 58979.75069477190846
                   RA J2000 = 09:53:09.3096
             RA J2000 (deg) = 148.28879        
                  Dec J2000 = 07:55:35.7499
            Dec J2000 (deg) = 7.9265972        
                  Tracking? = True
              Azimuth (deg) = 0
           Zenith Ang (deg) = 0
            Number of polns = 2 (summed)
           Sample time (us) = 5.12             
         Central freq (MHz) = 63.96484375      
          Low channel (MHz) = 45.3125          
         High channel (MHz) = 82.6171875       
        Channel width (MHz) = 0.1953125        
         Number of channels = 192
      Total Bandwidth (MHz) = 37.5             
                       Beam = 0 of 1
            Beam FWHM (deg) = 3.222
         Spectra per subint = 2400
           Spectra per file = 1799352
      Time per subint (sec) = 0.012288
        Time per file (sec) = 9.21268224
            bits per sample = 2
          bytes per spectra = 48
        samples per spectra = 192
           bytes per subint = 115200
         samples per subint = 460800
                zero offset = 0                
           Invert the band? = False
       bytes in file header = 349
```

We figured that this has probably something to do with the “worklen” variable in the “prepdata.c" code, or something related to the loading of the filterbank file in "sigproc_fb.c"

If you would like to give it a try, I copied the filterbank file that we used in our fortalesa server, for you to grab at:
```
/scratch1/dropfolder/nenufar_B0950+08_digifilled.fil
```
Note that the error occurs when we try to use DMs >= 40.6.
With any DMs lower than that value, prepdata works ok with the file that we are using in this example.

Would it be possible for you to fix this, so as to make PRESTO work with search-mode data taken at super-low frequencies, like those of NenuFar?

Thank you very much in advance!

Cheers,
Alessandro 

```
=============================================================================
aridolfi@nancep5:/data/aridolfi/B0950_test$ prepdata -nobary -dm 40.6 -o  ts_B0950+08_digifilled B0950+08_digifilled.fil


          Pulsar Data Preparation Routine
   Type conversion, de-dispersion, barycentering.
                by Scott M. Ransom

Assuming the data are SIGPROC filterbank format...
Reading SIGPROC filterbank data from 1 file:
 'B0950+08_digifilled.fil'

   Number of files = 1
      Num of polns = 2 (summed)
 Center freq (MHz) = 63.964844
   Num of channels = 192
   Sample time (s) = 5.12e-06      
    Spectra/subint = 2400
  Total points (N) = 1799352
    Total time (s) = 9.21268224    
    Clipping sigma = 6.000
  Invert the band? = False
         Byteswap? = False
    Remove zeroDM? = False

File  Start Spec   Samples     Padding        Start MJD
----  ----------  ----------  ----------  --------------------
1              0     1799352           0  58979.75069477190846

Setting a 'good' output length of 1920000 samples
Writing output data to 'ts_B0950+08_digifilled.dat'.
Writing information to 'ts_B0950+08_digifilled.inf'.

Massaging the data ...


Error in gen_fvect(): Cannot allocate memory
Amount Complete = 0%
aridolfi@nancep5:/data/aridolfi/B0950_test$
```

Version of PRESTO used:
```
commit 853b985957597cfc364b69e1bf26dc447ca13876 (HEAD -> master, origin/master, origin/HEAD)
Author: Scott Ransom <sransom@nrao.edu>
Date:   Sat Jan 9 16:06:37 2021 -0500

    Fixed bug when using -pfact/-ffact and have non zero fdot or fdotdot
```

**Comment by scottransom** [comment]
OK.  So I did a little investigation into this and there are two issues.

First off, the "Cannot allocate memory" happens because it is asking for a negative amount of RAM because the multiplication in the argument [here](https://github.com/scottransom/presto/blob/766fd27d0e6ff80d10d1e1ed6e575bea2d5b956f/src/backend_common.c#L537) is overflowing a regular integer.

Second, that is problematic even without the overflow, since if I fixed the overflow (using a long long, for instance), that line and the next would be asking for 16GB (at a minimum) of RAM.

The reason for this is that the dispersive delay between the top and bottom of the Nenufar data is longer than the duration of the file!  Since PRESTO uses a small multiple of the dispersive delay across the band as the "blocksize" to process the data, it is simply not able to do it with this file (even if it could allocate the RAM).

I'm not sure even what the right thing to do if you request to dedisperse such an filterbank file.  Because you have missing data at all times for an astrophysical source with a high-enough DM (for the example file, that is a DM of about 6.52!)

So I'm not sure where to go with this.  I'm not even sure that it is a bug, per se....

Interested to hear your thoughts.

That is problematic since
